velox/docs/functions/spark/aggregate.rst

```diff
@@ -107,13 +107,17 @@ General Aggregate Functions
 
     Returns the sum of `x`.
 
-    Supported types are TINYINT, SMALLINT, INTEGER, BIGINT, REAL and DOUBLE.
+    Supported types are TINYINT, SMALLINT, INTEGER, BIGINT, REAL, DOUBLE and DECIMAL.
 
     When x is of type DOUBLE, the result type is DOUBLE.
     When x is of type REAL, the result type is REAL.
+    When x is of type DECIMAL(p, s), the result type is DECIMAL(p + 10, s), where (p + 10) is capped at 38.
+
     For all other input types, the result type is BIGINT.
 
-    Note: When the sum of BIGINT values exceeds its limit, it cycles to the overflowed value rather than raising an error.
+    Note:
+    For DECIMAL type, when an overflow occurs in the accumulation, it returns null. For all other input types,
```

Comment:
I wonder why overflow for different types is handled differently. Is this how Spark behaves?

Also, wondering what's the result of sum(all-nulls). Is it 0? If so, it would be nice to clarify that as well. Is this 0 for all types, or only decimal?

Reply:
> Is this how Spark behaves?

Yes, this is the behaves of Spark.

> what's the result of sum(all-nulls)?
The result is null. For decimal, it is also null. I will add the explanation for null input.